lib/fluent/plugin/out_amplitude.rb

```diff
@@ -163,15 +166,27 @@ def extract_event_properties!(amplitude_hash, record)
     def send_to_amplitude(records)
       log.info("sending #{records.length} to amplitude")
       errors = []
+      records_sent = 0
       until records.empty?
         records_to_send = records.pop(500)
         start_time = Time.now.to_i
         res = AmplitudeAPI.track(records_to_send)
-        unless res.response_code == 200
+        @statsd.track('fluentd.amplitude.request_time', res.total_time * 1000)
+        if res.response_code == 200
+          @statsd.track(
+            'fluentd.amplitude.records_sent',
+            records_to_send.length
+          )
+          records_sent += records_to_send.length
+          log.info(
+            "Amplitude request complete. Duration: #{res.total_time * 1000}, Records: #{records_to_send.length}"
+          )
+        else
           fail_time = Time.now.to_i
           errors << [res.response_code, res.body, records_to_send, fail_time - start_time]
         end
       end
+      log.info("sent #{records_sent} to amplitude")
```

Comment:
can we also log this to statsd?

Reply:
That would double-track, since we're tracking after each batch on line 178.

Reply:
ok, sorry i see it now, so maybe track errors, too?